src/java.se/share/data/jdwp/jdwp.spec

```diff
@@ -1147,7 +1147,7 @@ JDWP "Java(tm) Debug Wire Protocol"
         "the method is being invoked if they were previously "
         "suspended by an event or by command. "
         "This is done to prevent the deadlocks "
-        "that will occur if any of the threads own monitors "
+        "that will occur if any of the threads own resources, such as monitors, "
```

Comment:
In passing, shouldn't that be "may occur" rather than the "will occur"?

Reply:
No, because the line below follows with "...that _will_ be needed by the invoked method". If the resource _will_ be needed, then the deadlock _will_ occur.